File: android/app/src/main/java/com/ccoan/ny/data/models/Department.kt
package com.ccoan.ny.data.models

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Department(
    val id: String,
    val name: String,
    val adminIds: List<String> = emptyList(),
    val memberIds: List<String> = emptyList(),
    val projectCount: Int? = null,
    val metadata: MetadataTimestamps? = null
)

@JsonClass(generateAdapter = true)
data class MetadataTimestamps(
    val createdAt: String,
    val updatedAt: String
)

@JsonClass(generateAdapter = true)
data class CreateDepartmentRequest(
    val name: String
)

@JsonClass(generateAdapter = true)
data class UpdateDepartmentRequest(
    val id: String,
    val name: String
)

@JsonClass(generateAdapter = true)
data class AddMemberRequest(
    val userId: String
)

File: ios/CCOAN/Models/Department.swift
import Foundation

struct Department: Codable, Identifiable {
    let id: String
    let name: String
    let adminIds: [String]
    let memberIds: [String]
    let metadata: MetadataTimestamps?

    enum CodingKeys: String, CodingKey {
        case id, name, adminIds, memberIds, metadata
    }

    init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)
        self.id = try container.decode(String.self, forKey: .id)
        self.name = try container.decode(String.self, forKey: .name)
        self.adminIds = try container.decodeIfPresent([String].self, forKey: .adminIds) ?? []
        self.memberIds = try container.decodeIfPresent([String].self, forKey: .memberIds) ?? []
        self.metadata = try container.decodeIfPresent(MetadataTimestamps.self, forKey: .metadata)
    }
}

struct MetadataTimestamps: Codable {
    let createdAt: String
    let updatedAt: String
}

struct CreateDepartmentRequest: Encodable {
    let name: String
}

struct UpdateDepartmentRequest: Encodable {
    let id: String
    let name: String
}

struct AddMemberRequest: Encodable {
    let userId: String
}
